pkg/executor/join/base_semi_join.go

```diff
@@ -129,3 +132,58 @@ func (b *baseSemiJoin) concatenateProbeAndBuildRows(joinedChk *chunk.Chunk, sqlK
 	b.finishCurrentLookupLoop(joinedChk)
 	return nil
 }
+
+// Only used for semi and anti semi join
+func (b *baseSemiJoin) generateResultChkForRightBuildNoOtherCondition(resultChk *chunk.Chunk) {
+	if len(b.offsets) == 0 {
+		return
+	}
+
+	for index, colIndex := range b.lUsed {
+		srcCol := b.currentChunk.Column(colIndex)
+		dstCol := resultChk.Column(index)
+		chunk.CopyRows(dstCol, srcCol, b.offsets)
+	}
+
+	if len(b.lUsed) == 0 {
+		resultChk.SetNumVirtualRows(resultChk.NumRows() + len(b.offsets))
+	} else {
+		resultChk.SetNumVirtualRows(resultChk.NumRows())
+	}
+}
+
+// Only used for semi and anti semi join
+func (b *baseSemiJoin) generateResultChkForRightBuildWithOtherCondition(remainCap int, chk *chunk.Chunk, resultRows []bool) {
+	for remainCap > 0 && (b.currentProbeRow < b.chunkRows) {
+		rowNumToTryAppend := min(remainCap, b.chunkRows-b.currentProbeRow)
+		start := b.currentProbeRow
+		end := b.currentProbeRow + rowNumToTryAppend
+
+		for index, usedColIdx := range b.lUsed {
+			dstCol := chk.Column(index)
+			srcCol := b.currentChunk.Column(usedColIdx)
+			chunk.CopySelectedRowsWithRowIDFunc(dstCol, srcCol, resultRows, start, end, func(i int) int {
+				return b.usedRows[i]
+			})
+		}
+
+		if len(b.lUsed) == 0 {
+			// For calculating virtual row num
+			virtualRowNum := chk.GetNumVirtualRows()
+			for i := start; i < end; i++ {
+				if b.isMatchedRows[i] {
```

Comment:
Although `resultRows` is the same as `b.isMatchedRows`, but I think you should use `if resultRows[i]` here. Or maybe you don't need pass `b.isMatchedRows` as an argument here.

Reply:
> Although `resultRows` is the same as `b.isMatchedRows`, but I think you should use `if resultRows[i]` here. Or maybe you don't need pass `b.isMatchedRows` as an argument here.

I replace `b.isMatchedRows` with `resultRows`.